util/secret/certificates.go

```diff
@@ -112,29 +113,53 @@ func NewCertificatesForInitialControlPlane(config *v1beta1.ClusterConfiguration)
 }
 
 // NewCertificatesForJoiningControlPlane gets any certs that exist and writes them to disk
-func NewCertificatesForJoiningControlPlane() Certificates {
-	return Certificates{
+func NewCertificatesForJoiningControlPlane(config *v1beta1.ClusterConfiguration) Certificates {
```

Comment:
This is a breaking change, we should deprecate this function in favor of a different one

Reply:
@vincepri  done
wondering if there is a way to avoid the `WithConfig` in the name of the new func once we deprecate this one...